drivers/platform/maxim/max32650/maxim_irq.c

```diff
@@ -388,16 +388,21 @@ int32_t max_irq_register_callback(struct no_os_irq_ctrl_desc *desc,
 		break;
 
 	case NO_OS_TIM_IRQ:
-		if (_events[NO_OS_EVT_RTC].actions == NULL) {
+		if (_events[NO_OS_EVT_TIM_ELAPSED].actions == NULL) {
```

Comment:
I would split this part in a separate commit, since it's a fix. Otherwise, it looks good.